drivers/ieee802154/Kconfig.cc2520

```diff
@@ -14,18 +14,21 @@ if IEEE802154_CC2520
 
 config IEEE802154_CC2520_DRV_NAME
 	string "TI CC2520 Driver's name"
+	depends on !HAS_DTS_SPI
```

Comment:
Under what situation after this PR will be able to set !HAS_DTS_SPI?  Seems like you should just remove this.